apispec_oneofschema/plugin.py

```diff
@@ -0,0 +1,58 @@
+# apispec-oneofschema - Plugin for apispec providing support for
+#                       Marshmallow-OneOfSchema schemas
+# Copyright (C) 2018  Tim Schumacher
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from apispec.ext import marshmallow
+from marshmallow_oneofschema import OneOfSchema
+
+class OneofOpenAPIConverter(marshmallow.OpenAPIConverter):
+    def schema2jsonschema(self, schema):
+        if self.openapi_version.major < 3 or not is_oneof(schema):
```

Comment:
Doesn't OAS2 allow this kind of inheritance?

From what I remember, only `discriminator` is new to OAS3, but inheritance is possible, although loosely defined.

Should the OAS version condition only apply to the line below that adds `discrimintator`?